lms/djangoapps/verify_student/views.py

```diff
@@ -921,6 +922,32 @@ def _send_email(user_id, subject, message):
     user.email_user(subject, message, from_address)
 
 
+def _set_user_requirement_status(attempt, namespace, status, reason=None):
+    """Sets the status of a credit requirement for the user,
+    based on a verification checkpoint.
+    """
+    checkpoint = None
+    try:
+        checkpoint = VerificationCheckpoint.objects.get(photo_verification=attempt)
+    except VerificationCheckpoint.DoesNotExist:
+        log.warn("Unable to find checkpoint for %s", attempt.user.username)
+
+    if checkpoint is not None:
+        course_key = checkpoint.course_id
+        credit_requirement = get_credit_requirement(
+            course_key, namespace, checkpoint.checkpoint_location
+        )
+        if credit_requirement is not None:
+            try:
+                set_credit_requirement_status(
+                    attempt.user.username, credit_requirement, status, reason
+                )
+            except Exception:  # pylint: disable=broad-except
+                # Catch exception if unable to add credit requirement
+                # status for user
+                log.warn("Unable to add Credit requirement status for %s", attempt.user.username)
```

Comment:
I have several concerns about this section:

1) As we discussed, public API functions should receive and return only JSON-serializable values, not Django models.  `get_credit_requirement()` above is returning a Django model, and `set_credit_requirement_status()` is receiving a Django model.  Why not just have `set_credit_requirement()` accept the username, namespace, and name, then set the status appropriately?

2) If an error occurs when setting the credit requirement status, we should log it with "ERROR" log level.  This would be a serious issue that would probably require manual intervention, so we'd want to know about it right away.

3) Please include user database IDs in log messages instead of usernames (slightly safer to avoid leaking PII).